nab/src/org/labkey/nab/NabManager.java

```diff
@@ -204,4 +215,26 @@ public void getDataPropertiesFromNabRunData(NabRunDataTable nabRunDataTable, Str
             throw new RuntimeSQLException(e);
         }
     }
+
+    /**
+     * Ensure the fit parameter value (as a JSON object) for the given NAb assay run specimen results row.
+     * If not already saved to the DB for the given NAbSpecimen row, save it.
+     * @param user
+     * @param assayRun
+     * @param specimenRow
+     * @param dilutionSummary
+     * @return CurveFit.Parameters as Map
+     * @throws FitFailedException
+     */
+    public Map<String, Object> ensureFitParameters(User user, NabSpecimen specimenRow, @Nullable DilutionAssayRun assayRun, @Nullable DilutionSummary dilutionSummary) throws FitFailedException

```

Comment:
Should we consider deprecating this function with a note that is mostly needed in the upgrade case and in 23.3.0 we can delete this since all upgrade cases that we support will have already been run?

Reply:
Yes, I'll add that note / tag.